interface-impl-amf/src/main/java/org/mule/amf/impl/model/ObjectParameterValidationStrategy.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.amf.impl.model;
+
+import amf.client.model.domain.AnyShape;
+import amf.client.validate.PayloadValidator;
+import amf.client.validate.ValidationReport;
+import static org.mule.amf.impl.model.MediaType.APPLICATION_YAML;
+
+public class ObjectParameterValidationStrategy implements ParameterValidationStrategy {
+
+
+  private final PayloadValidator parameterValidator;
+
+  ObjectParameterValidationStrategy(AnyShape anyShape){
```

Comment:
name Object is not clear, based in this condition (isArray() || isUnion()) , Composite is an option

Reply:
Scalar isn't true either, I've just leave those both for testing purpose, trying to think a better ones maybe performance related names